validator/telegram/telegram.go

```diff
@@ -0,0 +1,173 @@
+		telg := Telegram{base}
+		return &telg
+	}
+}
+
+func (telegram *Telegram) GeneratePostPayload() (post map[string]string) {
+	post = make(map[string]string, 0)
+	for lang_code, template := range POST_STRUCT {
+		post[lang_code] = fmt.Sprintf(template, telegram.Identity)
+	}
+
+	return post
+}
+
+func (telegram *Telegram) GenerateSignPayload() (payload string) {
+	telegram.Identity = strings.ToLower(telegram.Identity)
+	payloadStruct := validator.H{
+		"action":     string(telegram.Action),
+		"identity":   telegram.Identity,
+		"platform":   "telegram",
+		"prev":       nil,
+		"created_at": util.TimeToTimestampString(telegram.CreatedAt),
+		"uuid":       telegram.Uuid.String(),
+	}
+	if telegram.Previous != "" {
+		payloadStruct["prev"] = telegram.Previous
+	}
+
+	payloadBytes, err := json.Marshal(payloadStruct)
+	if err != nil {
+		l.Warnf("Error when marshaling struct: %s", err.Error())
+		return ""
+	}
+
+	return string(payloadBytes)
+}
+
+func (telegram *Telegram) Validate() (err error) {
+	initClient()
+	telegram.Identity = strings.ToLower(telegram.Identity)
+	telegram.SignaturePayload = telegram.GenerateSignPayload()
+	// Deletion. No need to fetch the telegram message.
+	if telegram.Action == types.Actions.Delete {
+		return mycrypto.ValidatePersonalSignature(telegram.SignaturePayload, telegram.Signature, telegram.Pubkey)
+	}
+
+	// Id of the direct message has been sent to the bot.
+	messageId, err := strconv.ParseInt(telegram.ProofLocation, 10, 64)
+	if err != nil {
+		return xerrors.Errorf("Error when parsing telegram message ID %s: %s", telegram.ProofLocation, err.Error())
+	}
+
+	if err := client.Run(context.Background(), func(ctx context.Context) error {
+
+		if _, err := client.Auth().Bot(ctx, config.C.Platform.Telegram.BotToken); err != nil {
+			return xerrors.Errorf("Error when authenticating the telegram bot: %v,", err)
+		}
+
+		msgsClass, err := client.API().MessagesGetMessages(ctx, []tg.InputMessageClass{
+			&tg.InputMessageID{
+				ID: int(messageId),
+			},
+		})
+
+		if err != nil {
+			return xerrors.Errorf("Error while fetching the user direct message (should be sent to the bot directly): %v,", err)
+		}
+
+		msgList, ok := msgsClass.(*tg.MessagesMessages)
+		if !ok || len(msgList.Messages) != 1 || len(msgList.Messages) != 2 {
+			return xerrors.New("Please try again sending an original message 1")
+		}
+		user, userOk := msgList.Users[0].(*tg.User)
+		if !userOk {
+			return xerrors.New("Please try again sending an original message 2")
+		}
+		if user.Bot {
+			user, userOk = msgList.Users[1].(*tg.User)
+
+		}
+		msg, msgOk := msgList.Messages[0].(*tg.Message)
+		if !msgOk || !userOk {
+			return xerrors.New("Please try again sending an original message 2")
+		}
+
+		if strings.EqualFold(user.Username, telegram.Identity) {
+			return xerrors.Errorf("Screen name mismatch: expect %s - actual %s", telegram.Identity, user.Username)
+		}
+		telegram.Text = msg.Message
+		telegram.AltID = strconv.FormatInt(user.ID, 10)
```

Comment:
Thank you for respecting our `AltID` design, just a trivial thing to concern:
 `AltID` here should refer to a user-modifiable, `@my_username` -like username, while `Identity` should refer to a immutable, `DigitUserIDGeneratedByTelegram` -like stuff.